common/render.go

```diff
@@ -136,44 +136,58 @@ func addCameraSystemOnce(w *ecs.World) {
 }
 
 func (rs *RenderSystem) Add(basic *ecs.BasicEntity, render *RenderComponent, space *SpaceComponent) {
-	// Setting default shader
-	if render.shader == nil {
-		switch render.Drawable.(type) {
-		case Triangle:
-			render.shader = LegacyShader
-		case Circle:
-			render.shader = LegacyShader
-		case Rectangle:
-			render.shader = LegacyShader
-		case ComplexTriangles:
-			render.shader = LegacyShader
-		case Text:
-			render.shader = TextShader
-		default:
-			render.shader = DefaultShader
+	if rs.EntityExists(basic) == -1 {
+		// Setting default shader
+		if render.shader == nil {
+			switch render.Drawable.(type) {
+			case Triangle:
+				render.shader = LegacyShader
+			case Circle:
+				render.shader = LegacyShader
+			case Rectangle:
+				render.shader = LegacyShader
+			case ComplexTriangles:
+				render.shader = LegacyShader
+			case Text:
+				render.shader = TextShader
+			default:
+				render.shader = DefaultShader
+			}
+		}
+
+		// This is to prevent users from using the wrong one
+		if render.shader == HUDShader {
+			switch render.Drawable.(type) {
+			case Triangle:
+				render.shader = LegacyHUDShader
+			case Circle:
+				render.shader = LegacyHUDShader
+			case Rectangle:
+				render.shader = LegacyHUDShader
+			case ComplexTriangles:
+				render.shader = LegacyHUDShader
+			case Text:
+				render.shader = TextHUDShader
+			default:
+				render.shader = HUDShader
+			}
 		}
+
+		rs.entities = append(rs.entities, renderEntity{basic, render, space})
+		rs.sortingNeeded = true
 	}
+}
 
-	// This is to prevent users from using the wrong one
-	if render.shader == HUDShader {
-		switch render.Drawable.(type) {
-		case Triangle:
-			render.shader = LegacyHUDShader
-		case Circle:
-			render.shader = LegacyHUDShader
-		case Rectangle:
-			render.shader = LegacyHUDShader
-		case ComplexTriangles:
-			render.shader = LegacyHUDShader
-		case Text:
-			render.shader = TextHUDShader
-		default:
-			render.shader = HUDShader
+// EntityExists looks if the entity is already into the System's entities. It will return the index >= 0 of the object into de rs.entities or -1 if it could not be found.
+func (rs *RenderSystem) EntityExists(basic *ecs.BasicEntity) int {
+	for index, entity := range rs.entities {
+		if entity.ID() == basic.ID() {
+			return index
+			break
```

Comment:
`break` is not needed after `return`, because we're already stopping executing the loop (due to `return`).